compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -141,14 +164,33 @@ public void perform(SyntaxNodeAnalysisContext ctx) {
         }
     }
 
-    private void validateRecordName(SyntaxNodeAnalysisContext ctx, TypeDefinitionNode typeDefinitionNode) {
-        String recordName = typeDefinitionNode.typeName().text().trim();
-        if (recordNames.contains(recordName)) {
-            reportDiagnosticInfo(ctx, typeDefinitionNode.location(), DiagnosticsCodes.PERSIST_119.getCode(),
-                    DiagnosticsCodes.PERSIST_119.getMessage(), DiagnosticsCodes.PERSIST_119.getSeverity());
+    private void checkPresenceOfEntityAnnotation(SyntaxNodeAnalysisContext ctx, NodeList<AnnotationNode> annotations,
+                                                 DiagnosticsCodes diagnosticsCodes) {
+        for (AnnotationNode annotation : annotations) {
+            if (annotation.annotReference().toSourceCode().trim().equals(Constants.ENTITY)) {
+                reportDiagnosticInfo(ctx, annotation.location(), diagnosticsCodes.getCode(),
+                        diagnosticsCodes.getMessage(),
+                        diagnosticsCodes.getSeverity());
+            }
```

Comment:
If this is sucessful there should be a break in processing rest of the annotation isnt it? Or is there a possibility of having same annotaion attached twice?